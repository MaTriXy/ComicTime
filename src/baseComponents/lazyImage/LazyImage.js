import React from "react";
import ProgressiveImage from "react-progressive-bg-image";
import LazyLoad from "react-lazy-load";

const style = {
  backgroundSize: "cover",
  backgroundPosition: "center center"
};

class LazyImage extends React.Component {
  render() {
    const height = this.props.height;
    return (
      <LazyLoad height={height} offsetTop={height} throttle={100}>
        <ProgressiveImage
          src={this.props.src}
          placeholder={this.props.placeholder}
          alt={this.props.alt}
          style={style}
        />
      </LazyLoad>
    );
  }
}

export default LazyImage;
