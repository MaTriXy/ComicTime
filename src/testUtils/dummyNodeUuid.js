jest.mock("node-uuid", () => {
  return {
    v4() {
      return "uuid";
    }
  };
});
