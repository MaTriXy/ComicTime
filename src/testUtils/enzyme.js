import React from "react";
import Enzyme from "enzyme";
import { createSerializer } from "enzyme-to-json";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { MuiThemeProvider } from "material-ui";
import comicTimeTheme from "../theme";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
Enzyme.configure({ adapter: new Adapter() });

export function mockHistory() {
  return {
    go: jest.fn(),
    goBack: jest.fn()
  };
}

export function translator(key) {
  return key;
}

export function mountWithMuyThemeProvider(component) {
  return Enzyme.mount(
    <MuiThemeProvider muiTheme={comicTimeTheme}>{component}</MuiThemeProvider>
  );
}

export function mountComponent(component) {
  return Enzyme.mount(component);
}

export function mountComponentAsJson(component) {
  return toJson(mountComponent(component));
}

export function shallowComponent(component) {
  return Enzyme.shallow(component);
}

export function shallowComponentAsJson(component) {
  return toJson(shallowComponent(component));
}

export function json(component) {
  return toJson(component);
}
