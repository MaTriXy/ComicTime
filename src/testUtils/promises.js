export async function flushPromises() {
  return new Promise(resolve => setImmediate(resolve));
}
