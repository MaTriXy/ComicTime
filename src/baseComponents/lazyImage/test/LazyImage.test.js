import React from "react";
import LazyImage from "../LazyImage";
import { mountComponentAsJson } from "../../../testUtils/enzyme";

describe("LazyImage", () => {
  it("renders the image configured using the configured parameters", () => {
    const component = mountComponentAsJson(
      <LazyImage
        height={200}
        src={"https://images.com/1"}
        placeholder={"https://images.com/2"}
        alt={"Alternative text"}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
