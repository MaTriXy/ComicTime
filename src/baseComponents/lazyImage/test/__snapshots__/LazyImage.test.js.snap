// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyImage renders the image configured using the configured parameters 1`] = `
<LazyImage
  alt="Alternative text"
  height={200}
  placeholder="https://images.com/2"
  src="https://images.com/1"
>
  <LazyLoad
    debounce={true}
    elementType="div"
    height={200}
    offset={0}
    offsetBottom={0}
    offsetHorizontal={0}
    offsetLeft={0}
    offsetRight={0}
    offsetTop={200}
    offsetVertical={0}
    throttle={100}
  >
    <div
      className="LazyLoad"
      style={
        Object {
          "height": 200,
          "width": undefined,
        }
      }
    />
  </LazyLoad>
</LazyImage>
`;
